app/src/examples/RuntimeTests/tests/core/useAnimatedStyle/reuseAnimatedStyle.test.tsx

```diff
@@ -0,0 +1,144 @@
+import { useEffect } from 'react';
+import { View, StyleSheet } from 'react-native';
+import Animated, { useSharedValue, useAnimatedStyle, withTiming, AnimatableValueObject } from 'react-native-reanimated';
+import React from 'react';
+import { ComparisonMode } from '../../../ReanimatedRuntimeTestsRunner/types';
+import {
+  describe,
+  test,
+  expect,
+  render,
+  useTestRef,
+  getTestComponent,
+  wait,
+} from '../../../ReanimatedRuntimeTestsRunner/RuntimeTestsApi';
+import { getComparisonModeForProp } from '../../../ReanimatedRuntimeTestsRunner/Comparators';
+
+describe('Test reusing animatedStyles', () => {
+  const COMPONENT_REF = {
+    ONE: 'ONE',
+    TWO: 'TWO',
+    THREE: 'THREE',
+  };
+
+  const TripleComponent = ({
+    startStyle,
+    finalStyle,
+    animate = false,
+  }: {
+    startStyle: AnimatableValueObject;
+    finalStyle: AnimatableValueObject;
+    animate: boolean;
+  }) => {
+    const toggle = useSharedValue(true);
+
+    const svStyle = useSharedValue(startStyle);
+
+    const refOne = useTestRef(COMPONENT_REF.ONE);
+    const refTwo = useTestRef(COMPONENT_REF.TWO);
+    const refThree = useTestRef(COMPONENT_REF.THREE);
+
+    const animatedStyle = useAnimatedStyle(() => {
+      if (!animate) {
+        return toggle.value ? startStyle : finalStyle;
+      } else {
+        return svStyle.value;
+      }
+    });
+
+    useEffect(() => {
+      if (!animate) {
+        toggle.value = false;
+      } else {
+        svStyle.value = withTiming(finalStyle, { duration: 900 });
```

Comment:
Is any reason why this animation is longer than standard 300ms? I'm asking because for more test cases it can take a lot of time.